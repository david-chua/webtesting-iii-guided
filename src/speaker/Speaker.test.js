import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import renderer from 'react-test-renderer';

import Speaker from './Speaker.js';

describe('<Speaker /> ', () => {
  xit ('should call the speak function on click', () =>{
    const speakMock = jest.fn();

    const { getByText } = render(<Speaker speak={speakMock} />);

    const speakButton = getByText(/speak/i);

    fireEvent.click(speakButton);

    expect(speakMock).toHaveBeenCalled();
  });

describe('<Speaker />', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<Speaker />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match the snapshot with a message', () => {
    const tree = renderer.create(<Speaker message={"hello"}/>).toJSON();

    expect(tree).toMatchSnapshot();

  })
})



});
