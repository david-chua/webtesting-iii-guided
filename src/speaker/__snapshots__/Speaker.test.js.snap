// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Speaker />  <Speaker /> should match the snapshot 1`] = `
<div>
  <button>
    Speak
  </button>
  <div />
</div>
`;

exports[`<Speaker />  <Speaker /> should match the snapshot with a message 1`] = `
<div>
  <button>
    Speak
  </button>
  <div>
    hello
  </div>
</div>
`;
